### DISCUSSION TITLE
Accessing model data in repeater

### DISCUSSION THREAD

**Discussion by avinash403** [area17/twill#2466] [Q&A]
### Discussed in https://github.com/area17/twill/discussions/2333

<div type='discussions-op-text'>

<sup>Originally posted by **avinash403** September 13, 2023</sup>
I have this parent model, let's say `Post` and child element with one to many relation let's say `Collaborator`. On my form, I have a list if collaborators added as a repeater to give functionality to create collaborators quickly. I want to put up a link to edit the collaborator on each of the repeater element. Is there a way to get the model data attached to a repeater element in the repeater blade file?

```
@twillRepeaterTitle('Collaborator')
@twillRepeaterTrigger('Add Collaborator')
@twillRepeaterGroup('app')

<x-twill::medias
    name="profile_picture"
    label="Profile Picture"
/>
<x-twill::input
    name="full_name"
    label="Full Name"
    :required="true"
/>

<a href="" {{-- here's where I need the model data to create the edit url --}}
   class="button button--action" style="float: right;">Edit</a>

```
</div>

**Comment by shaifulborhan** [comment]
I would like to know how to do this also.

**Comment by avinash403** [comment]
I wrote a script to handle that using IDs in the repeater. I added `repeater-edit-link` in `Repeater.vue` (Or you can create a custom repeater component and use that)
```
        <a17-button :id="'edit-link-container'+block.id" slot="block-actions" variant="icon" class="hidden">
              <a :id="'edit-link-'+block.id" target="_blank" class="repeater-edit-link">
                  <span v-svg symbol="edit"></span>
              </a>
          </a17-button>
```
```
        function updateHref() {
            let twillLinks = document.getElementsByClassName('repeater-edit-link');
            for(let i = 0; i < twillLinks.length; i++) {
                let divId = twillLinks[i].id;
                let identifier = divId.match(/\d+/g)[0];
                twillLinks[i].setAttribute('href', '/admin/categories/' + identifier + '/edit');
                twillLinks[i].parentElement.classList.remove('hidden');
            }
        }
```
